src/software/simulation/physics/simulation_contact_listener.cpp

```diff
@@ -33,6 +33,17 @@ void SimulationContactListener::BeginContact(b2Contact *contact)
                 contact_callback(robot, ball);
             }
         }
+        if (auto ball_field_pair = isBallFieldWallContact(user_data_a, user_data_b))
+        {
+            // If the ball hits a wall it's out of bounds, so we stop it to make it easier
+            // to retrieve for ball placement (otherwise it will bounce around the field)
+            PhysicsBall *ball = ball_field_pair->first;
+            // Apply an impulse to nearly stop the ball. It's preferable if it bounces
+            // away from a wall very slightly for realism, and it should make it easier to
+            // retrieve for ball placement
+            ball->applyImpulse(
+                -ball->momentum().normalize(ball->momentum().length() * 0.95));
```

Comment:
something less aggressive feels more realistic, e.g.
```suggestion
                -ball->momentum().normalize(ball->momentum().length() * 0.8));
```

This should be defined as a constant in the header

Reply:
done